app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt

```diff
@@ -195,12 +209,97 @@ class VerifyPhoneActivity : AppCompatActivity(), ConnectionUtils.SmsVerification
+                        val loginResponse = customStub.login(LoginRequest.newBuilder().setEmail(email).setPassword(password).build())
+
+                        uiThread {
+
+                            if (loginResponse.statusCode == LoginResponse.StatusCode.OK && !loginResponse.user.isBlocked) {
+
+                                MiscellaneousUtils.sessionId = loginResponse.sessionId
+
+                                val intent = Intent(this@VerifyPhoneActivity, MainActivity::class.java)
+                                intent.putExtras(getIntent())
+                                with(intent) {
+                                    putExtra(MainActivity.CASH_WORTH_KEY, loginResponse.user.cash)
+                                    putExtra(MainActivity.TOTAL_WORTH_KEY, loginResponse.user.total)
+                                    putExtra(MainActivity.RESERVED_CASH_KEY, loginResponse.user.reservedCash)
+                                }
+                                startActivity(intent)
+                                finish()
+                            }
+                        }
+                    }
+                         else {
+                            uiThread { startLoginActivity() }
+                        }
+                    } else {
+                        val snackBar = Snackbar.make(findViewById<View>(android.R.id.content), resources.getString(R.string.error_server_down), Snackbar.LENGTH_INDEFINITE)
+                                .setAction("RETRY") {
+                                    startLoginProcess(email, password)
+                                }
+
+                        snackBar.setActionTextColor(ContextCompat.getColor(this@VerifyPhoneActivity, R.color.neon_green))
+                        snackBar.view.setBackgroundColor(Color.parseColor("#20202C"))
+                        snackBar.show()
+
+                    }
+
+                } else /* No internet available */ {
+                    uiThread {
+
+                        val snackBar = Snackbar.make(findViewById<View>(android.R.id.content), "Please check internet connection", Snackbar.LENGTH_INDEFINITE)
+                                .setAction("RETRY") {
+                                    startLoginProcess(email, password)
+
+                                }
+
+                        snackBar.setActionTextColor(ContextCompat.getColor(this@VerifyPhoneActivity, R.color.neon_green))
+                        snackBar.view.setBackgroundColor(Color.parseColor("#20202C"))
+                        snackBar.show()
+
+                    }
+                }
+            }
+        }
+
+    private fun startLoginProcess(email: String?, password: String?) {
+
+        if (email != null && password != null && email != "") {
+            loginAsynchronously(email, password)
+        } else {
+            startLoginActivity()
+        }
+    }
```

Comment:
Eliminate this function, and use this body phoneVerificationSuccessful() function